pkg/operator/mon/health.go

```diff
@@ -143,9 +143,64 @@ func (c *Cluster) checkHealth() error {
 		return nil
 	}
 
+	// check if there are more than two mons running on the same node, failover one mon in that case
+	done, err := c.checkMonsOnSameNode()
+	if done || err != nil {
+		return err
+	}
+
+	done, err = c.checkMonsOnValidNodes()
+	if done || err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func (c *Cluster) checkMonsOnSameNode() (bool, error) {
+	nodesUsed := map[string]struct{}{}
+	for name, node := range c.mapping.Node {
+		// when the node is already in the list we have more than one mon on that node
+		if _, ok := nodesUsed[node.Name]; ok {
+			// get list of available nodes for mons
+			availableNodes, _, err := c.getAvailableMonNodes()
+			if err != nil {
+				return true, fmt.Errorf("failed to get available mon nodes. %+v", err)
+			}
+			// if there are enough nodes for mons failover the mon too much from the node
+			if len(availableNodes) > 0 {
+				logger.Infof("rebalance: enough nodes available %d to failover mon %s", len(availableNodes), name)
+				c.failMon(len(c.clusterInfo.Monitors), name)
+			} else {
+				logger.Warningf("rebalance: not enough nodes available to failover mon %s", name)
```

Comment:
could this be a debug statement instead? A node could be down for a long time so we will get this every 20sec.

Also running in test cluster like minikube, we would get this every 20s since it will only have one node for 3 mons.

Reply:
Moved it to `debug` level.